librz/core/rop.c

```diff
@@ -358,59 +394,127 @@ RZ_IPI RzRopRegInfo *rz_core_rop_reg_info_dup(RzRopRegInfo *src) {
 	return dup;
 }
 
-static bool is_stack_pointer(const RzCore *core, const char *name) {
-	RzRegItem *reg_item = rz_reg_get(core->analysis->reg, name, RZ_REG_TYPE_GPR);
-	if (!reg_item) {
-		return false;
-	}
-	if (core->analysis->bits == 32 && RZ_STR_EQ(core->analysis->cpu, "x86")) {
-		return !strcmp(reg_item->name, "esp");
-	}
-	if (core->analysis->bits == 64) {
-		return !strcmp(reg_item->name, "rsp") || !strcmp(reg_item->name, "esp");
-	}
-	if (core->analysis->bits == 64 && !strcmp(core->analysis->cpu, "arm")) {
-		return !strcmp(reg_item->name, "r13");
+/**
+ * \brief Check if a register with a specific name exists in the modified registers of a RzRopGadgetInfo object.
+ * \param gadget_info RZ_NONNULL Pointer to the RzRopGadgetInfo object.
+ * \param name RZ_NONNULL Pointer to the name of the register.
+ * \return true if a register with the given name exists, false otherwise.
+ *
+ * Checks the modified registers in the RzRopGadgetInfo object to see if a register with the given name exists.
+ */
+RZ_API bool rz_core_rop_gadget_info_has_register(const RZ_NONNULL RzRopGadgetInfo *gadget_info, RZ_NONNULL const char *name) {
+	rz_return_val_if_fail(gadget_info && name, false);
+	void **it;
+	rz_pvector_foreach (gadget_info->modified_registers, it) {
+		const RzRopRegInfo *reg_info = *it;
+		if (RZ_STR_EQ(reg_info->name, name)) {
+			return true;
+		}
 	}
-	return reg_item->name;
+	return false;
 }
 
-static bool is_base_pointer(const RzCore *core, const char *name) {
-	RzRegItem *reg_item = rz_reg_get(core->analysis->reg, name, RZ_REG_TYPE_GPR);
-	if (!reg_item) {
-		return false;
+static bool is_var_write_event(const RzRopRegInfo *reg_info) {
+	return reg_info->is_var_write;
+}
+
+static bool is_mem_read_event(const RzRopRegInfo *reg_info) {
+	return reg_info->is_mem_read;
+}
+
+static bool is_mem_write_event(const RzRopRegInfo *reg_info) {
+	return reg_info->is_mem_write;
+}
+
+static bool is_pc_write_event(const RzRopRegInfo *reg_info) {
+	return reg_info->is_pc_write;
+}
+
+event_check_fn event_functions[RZ_ROP_EVENT_COUNT] = {
```

Comment:
I am sorry, oversaw it last time. Please make it static. `event_functions` might be used as name somewhere else.

Reply:
Agree :+1: